gui/src/pages/Process.svelte

```diff
@@ -62,88 +64,81 @@
   });
 </script>
 
-<Layout>
-  <section>
-    {#if process}
-      <div>
-        <div id="heading">
-          <button
-            class="back-button"
-            on:click={() =>
-              void router.push(
-                `#/workspaces/${encodeURIComponent(workspaceId)}/`,
-              )}
-          >
-            🠔 Back
-          </button>
-          <h1>{process.name}</h1>
-        </div>
-        <h3>Status</h3>
-        {#if process.running}
-          <table>
-            <tr>
-              <td>Status</td>
-              <td>{process.running ? 'Running' : 'Stopped'}</td>
-            </tr>
-            <tr>
-              <td>CPU</td>
-              <td>{process.cpuPercent.toFixed(2)}%</td>
-              <td
-                ><svg
-                  bind:this={sparklineSvg}
-                  class="sparkline"
-                  width="100"
-                  height="30"
-                  stroke-width="3"
-                /></td
-              >
-            </tr>
-            <tr>
-              <td>Resident Memory</td>
-              <td><BytesLabel value={process.residentMemory} /></td>
-            </tr>
-            <tr>
-              <td>Started at</td>
-              <td
-                ><span title={new Date(process.createTime).toISOString()}
-                  >{new Date(process.createTime).toLocaleTimeString()}</span
-                ></td
-              >
-              <td
-                ><svg
-                  class="sparkline"
-                  width="100"
-                  height="30"
-                  stroke-width="3"
-                /></td
-              >
-            </tr>
-            <tr>
-              <td>Local Ports</td>
-              <td>{process.ports?.join(', ') ?? 'None'}</td>
-            </tr>
-            <tr>
-              <td>Children</td>
-              <td>{process.childrenExecutables?.join(', ') ?? 'None'}</td>
-            </tr>
-          </table>
-          <h3>Environment</h3>
-          <table>
-            {#each Object.entries(process.envVars ?? {}) as [name, val] (name)}
+<Layout showBackButton backButtonRoute={workspaceRoute}>
```

Comment:
I'd suggest cleaning up the props a bit so that the back button is displayed iff `backButtonRoute` is supplied rather than having separate props for (en|dis)abling and the route itself. It does require that we always be explicit with the route, but I think that is a good thing.

Reply:
Good idea